### PR TITLE
Fix Ogg broadcasting

### PR DESCRIPTION
**Pull request by JosepMaJAZ**
Ogg vorbis does not work for Icecast. This bug has come from PR #2415 because on Broadcasting, the Ogg vorbis plugin has the code "OggVorbis", while the recording code is "OGG".

### CODE REVIEW

**Comment by Be-ing** [comment]
Please make a separate PR for each of those bugs.

**Comment by Be-ing** [comment]
Should this be targeted at 2.2?

**Comment by JosepMaJAZ** [comment]
The Ogg bug is 2.3 only.
The other two seems that people is reporting them on 2.2.

So... Should I simply leave this for the ogg fix and make the other changes over 2.2?

**Comment by Be-ing** [comment]
yes

**Comment by Holzhaus** [comment]
Please don't forget to add PRs to milestones and the project board.

File: src/preferences/broadcastsettings_legacy.cpp [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #include "preferences/broadcastsettings.h"
+#include "broadcast/defs_broadcast.h"
 
+#define BROADCAST_FORMAT_OV_LEGACYNAME "OggVorbis"
```
Comment (uklotzde) [inline src/preferences/broadcastsettings_legacy.cpp:4]: We should not define this constant twice.

In this case it should be moved into the header file with a comment why it is needed and that it shall only be used for this special purpose.

File: src/broadcast/defs_broadcast.h [hunk 2]
```diff
@@ -23,8 +23,10 @@
 #define BROADCAST_BITRATE_48KBPS 48
 #define BROADCAST_BITRATE_32KBPS 32
 
+// WARNING! def_recordings and defs_broadcast format codes have to be the same
```
Comment (uklotzde) [inline src/broadcast/defs_broadcast.h:26]: Instead of adding a warning: How about defining them only once thereby fixing the root cause?

**Review by uklotzde — CHANGES_REQUESTED** [review]

File: src/broadcast/defs_broadcast.h (hunk 2 above)
Comment (JosepMaJAZ) [reply src/broadcast/defs_broadcast.h]: done

**Comment by uklotzde** [comment]
Did you forget to commit the remaining changes?

**Review by uklotzde — APPROVED** [review]

**Comment by uklotzde** [comment]
LGTM. Thank you!

**Pull request merged**